domain/importtracker.go

```diff
@@ -50,12 +60,17 @@ func (ir *ImportTracker) Add(importStr string, as string, global bool) error {
 	}
 	importStr = parsedImport.ProjectCanonical() // normalize
 	var path string
+	allowPrivileged := currentlyPrivileged
 	if parsedImport.IsImportReference() {
 		return errors.Errorf("IMPORT %s not supported", importStr)
 	} else if parsedImport.IsRemote() {
 		path = parsedImport.GetGitURL()
+		allowPrivileged = allowPrivileged && allowPrivilegedFlag
 	} else if parsedImport.IsLocalExternal() {
 		path = parsedImport.GetLocalPath()
+		if allowPrivilegedFlag {
+			ir.console.Printf("the --allow-privileged flag has no effect when referencing a local or relative target\n")
```

Comment:
Nit: Relative targets are also local (internal vs external is the naming I've used before for same earthfile vs different earthfile)

```suggestion
			ir.console.Printf("the --allow-privileged flag has no effect when referencing a local target\n")
```

Reply:
good point :)